extensions/interactions/NumberWithUnits/directives/number-with-units-interaction.component.html

```diff
@@ -1,14 +1,21 @@
-<form autocomplete="off" ng-disabled="$ctrl.NumberWithUnitsForm.$invalid" ng-submit="$ctrl.submitAnswer($ctrl.answer)" role="form" class="form-horizontal">
+<form autocomplete="off" [disabled]="!isValid ? true : false" role="form" class="form-horizontal">
   <!-- This ng-form is needed for submitting answer on hitting enter. -->
-  <ng-form name="$ctrl.NumberWithUnitsForm" class="input">
-    <input type="text" name="answer" ng-model="$ctrl.answer" class="form-control">
-  </ng-form>
+  <input type="text"
+         name="answer"
+         [(ngModel)]="answer"
+         (ngModelChange)="answerValueChanged()"
+         class="form-control"
+         (keyup.enter)="submitAnswer()"
```

Comment:
Why `keyup` event is used instead of `(submit)`?

Reply:
Done Now. Move `submitAnswer()` to form.